test-perf/GeneratorBenchmark.ts

```diff
@@ -0,0 +1,86 @@
+import "mocha";
+import * as assert from "power-assert";
+
+import { ActionResult, successOn } from "../src/action/ActionResult";
+import { GitHubRepoRef } from "../src/operations/common/GitHubRepoRef";
+import { ProjectOperationCredentials } from "../src/operations/common/ProjectOperationCredentials";
+import { RepoId } from "../src/operations/common/RepoId";
+import { generate, ProjectPersister } from "../src/operations/generate/generatorUtils";
+import { LocalProject } from "../src/project/local/LocalProject";
+import { NodeFsLocalProject } from "../src/project/local/NodeFsLocalProject";
+import { InMemoryProject } from "../src/project/mem/InMemoryProject";
+import { Project } from "../src/project/Project";
+
+describe("generator benchmark", () => {
+
+    const numberOfTimesToGenerate = 500;
+    const numberOfSeedProjectFiles = 100;
+    const numberOfLipsumInFile = 10;
+
+    it("should create a new GitHub repo using GenericGenerator", done => {
+        const repoName = "perfTestRepo";
+
+        const seed = constructProject(numberOfSeedProjectFiles, numberOfLipsumInFile);
+        generateFromSeed(seed, repoName)
+        .then(r => {
+            const result = r as ActionResult<LocalProject>;
+            NodeFsLocalProject.fromExistingDirectory(new GitHubRepoRef("atomist", repoName), result.target.baseDir)
+            .then(created => {
+                created.totalFileCount().then(fileCount =>
+                    assert.deepEqual(fileCount, numberOfSeedProjectFiles),
```

Comment:
Is `deepEqual` necessary here?

Reply:
yes, I think it is always necessary.

Reply:
I think it is required because the package.json `"directory": { "test": "test" }` means espower-typescript does not look outside the test directory, and this test resides outside it.